src/public/lib/nbgv-python/src/nbgv_python/runner.py

```diff
@@ -46,7 +46,7 @@ def run(
 
     def get_version(self, project_dir: Path | str = ".") -> GitVersion:
         """Return version metadata for *project_dir*."""
-        project_path = Path(project_dir)
+        project_path = Path(project_dir).resolve(strict=False)
```

Comment:
Nice fix making --project absolute and keeping cwd unchanged. One small edge case: Path(...).resolve(strict=False) will not expand '~'. Consider Path(project_dir).expanduser().resolve(strict=False) so user-supplied paths like ~/repo work consistently.